src/Microsoft.TestPlatform.TestHostProvider/Hosting/DefaultTestHostManager.cs

```diff
@@ -186,11 +189,27 @@ public Task<bool> LaunchTestHostAsync(TestProcessStartInfo testHostStartInfo, Ca
         EqtTrace.Verbose("DefaultTestHostmanager: Full path of {0} is {1}", testHostProcessName, testhostProcessPath);
 
         var launcherPath = testhostProcessPath;
-        if (!_environment.OperatingSystem.Equals(PlatformOperatingSystem.Windows) &&
-            !_processHelper.GetCurrentProcessFileName()!.EndsWith(DotnetHostHelper.MONOEXENAME, StringComparison.OrdinalIgnoreCase))
+        var processName = _processHelper.GetCurrentProcessFileName();
+        if (processName is not null)
         {
-            launcherPath = _dotnetHostHelper.GetMonoPath();
-            argumentsString = testhostProcessPath.AddDoubleQuote() + " " + argumentsString;
+            if (!_environment.OperatingSystem.Equals(PlatformOperatingSystem.Windows) &&
+                !processName.EndsWith(DotnetHostHelper.MONOEXENAME, StringComparison.OrdinalIgnoreCase))
```

Comment:
Sorry I missed this nit (feel free to ignore it if that's only comment left). Just for the sake of uniformity as you are using them at the front just below.

```suggestion
            if (!_environment.OperatingSystem.Equals(PlatformOperatingSystem.Windows)
                && !processName.EndsWith(DotnetHostHelper.MONOEXENAME, StringComparison.OrdinalIgnoreCase))
```

Reply:
Fixed.